pennylane/measurements/measurements.py

```diff
@@ -676,6 +676,22 @@ def process_state(self, state: Sequence[complex], wire_order: Wires):
                 dimension :math:`2^n` acts on a subspace of :math:`n` wires
         """
 
+    def process_density_matrix(self, density_matrix: Sequence[complex], wire_order: Wires):
+        """
+        Process the given density matrix.
+
+        Args:
+            density_matrix (Sequence[complex]): The density matrix representing the (mixed) quantum state,
```

Comment:
Is `Sequence[complex]` right here? :thinking:
I was expecting `Union[array, Sequence[array]]`.

Reply:
This is in agreement with how we type hint the state in `process_state`.  

We could probably update this to `TensorLike`, but I'm also fine with it the way it is.  We do have an upcoming push to improve how we type hint things, so we could look into improving how we type hint and document `TensorLike` objects as well.